pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -166,6 +189,198 @@ func TestReconcileClusterPool(t *testing.T) {
+					testcd.WithCustomization("test-cdc-1"),
+					testcd.Broken(),
+				),
+			},
+			inventory:                    true,
+			expectedTotalClusters:        0,
+			expectedObservedSize:         1,
+			expectedObservedReady:        0,
+			expectedInventoryVaildStatus: corev1.ConditionFalse,
+			expectedInventoryMessage:     map[string][]string{"BrokenByCloud": {"test-cdc-1"}},
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedAssignedCDCs:         1,
+		},
+		{
+			name: "cp with inventory and good cdc is valid, cd created",
+			existing: []runtime.Object{
+				inventoryPoolBuilder.Build(testcp.WithSize(1)),
+				testcdc.FullBuilder(testNamespace, "test-cdc-1", scheme).Build(),
+				unclaimedCDBuilder("c1").Build(
+					testcd.WithCustomization("test-cdc-1"),
+					testcd.Running(),
+				),
+			},
+			inventory:                    true,
+			expectedTotalClusters:        0,
+			expectedObservedSize:         1,
+			expectedObservedReady:        1,
+			expectedInventoryVaildStatus: corev1.ConditionTrue,
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedAssignedCDCs:         1,
+		},
+		{
+			// Second prioritize by last used
+			name: "cp with inventory - correct prioritization - same status",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(
+					testcp.WithSize(1),
+					testcp.WithInventory([]string{"test-cdc-successful-old", "test-cdc-unused-new"}),
+				),
+				testcdc.FullBuilder(testNamespace, "test-cdc-successful-old", scheme).Build(
+					testcdc.WithLastStatus(hivev1.LastApplySucceeded),
+					testcdc.WithLastTime(nowish.Add(-time.Hour)),
+				),
+				testcdc.FullBuilder(testNamespace, "test-cdc-unused-new", scheme).Build(
+					testcdc.WithLastTime(nowish),
+				),
+			},
+			expectedTotalClusters:        1,
+			expectedInventoryVaildStatus: corev1.ConditionTrue,
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedInventoryNext:        "test-cdc-successful-old",
+			expectedAssignedCDCs:         1,
+		},
+		{
+			// First prioritize unused/successful vs broken (cloud/syntax)
+			name: "cp with inventory - correct prioritization - successful vs broken",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(
+					testcp.WithSize(1),
+					testcp.WithInventory([]string{"test-cdc-successful-new", "test-cdc-broken-old"}),
+				),
+				testcdc.FullBuilder(testNamespace, "test-cdc-broken-old", scheme).Build(
+					testcdc.WithLastStatus(hivev1.LastApplySucceeded),
+					testcdc.WithLastTime(nowish.Add(-time.Hour)),
+				),
+				testcdc.FullBuilder(testNamespace, "test-cdc-successful-new", scheme).Build(
+					testcdc.WithLastStatus(hivev1.LastApplyBrokenCloud),
+					testcdc.WithLastTime(nowish),
+				),
+			},
+			expectedTotalClusters:        1,
+			expectedInventoryVaildStatus: corev1.ConditionFalse,
+			expectedPoolVersion:          "06983eaafac7f695",
+			expectedInventoryNext:        "test-cdc-successful-new",
```

Comment:
Wait, shouldn't the succeeded one come out first? (And shouldn't we make the broken one older than the succeeded one, to prove that we're paying attention to the status over the age?)